pkg/operator/workload_controller_openshiftapiserver_v311_00.go

```diff
@@ -81,25 +82,51 @@ func syncOpenShiftAPIServer_v311_00_to_latest(c OpenShiftAPIServerOperator, orig
 
 	// only manage the apiservices if we have ready pods for the daemonset.  This makes sure that if we're taking over for
 	// something else, we don't stomp their apiservices until ours have a reasonable chance at working.
+	var actualAPIServices []*apiregistrationv1.APIService
 	if actualDaemonSet != nil && actualDaemonSet.Status.NumberReady > 0 {
-		err = manageAPIServices_v311_00_to_latest(c.apiregistrationv1Client)
+		actualAPIServices, err = manageAPIServices_v311_00_to_latest(c.apiregistrationv1Client)
 		if err != nil {
 			errors = append(errors, fmt.Errorf("%q: %v", "apiservices", err))
 		}
 	}
 
 	// manage status
-	if actualDaemonSet.Status.NumberAvailable > 0 {
+	var availableConditionReason string
+	var availableConditionMessages []string
+	if actualDaemonSet != nil && actualDaemonSet.Status.NumberAvailable == 0 {
```

Comment:
Seems like this should be an OR

Reply:
The case where `actualDaemonSet == nil` is handle above.